### DISCUSSION TITLE
Do wifi drivers use CPU in the background?

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#14393] [Libraries & Drivers]
Do wifi drivers (specifically, the Murata 1DX on PYBD, but more generally, Cypress/Infineon wifi chip on STM32 MCUs) consume any CPU outside of calls to the `network` module? 

I ask as we are seeing some devices running the same code, behave differently with respect to async code. The most obvious difference between the devices is they are in different wifi environments

**Comment by peterhinch** [comment]
The `network` module is synchronous and the STM ports are bare metal ports. This doesn't rule out the possibility that there might be ISR's running, but I've never seen any suggestion that this is actually the case.

Asynchronous applications can be prone to non-deterministic behaviour, notably if a coroutine sometimes hogs the processor. Occasional CPU hogging can be difficult to isolate, but there are some debugging aids for `asyncio` listed [in this doc](https://github.com/peterhinch/micropython-async/blob/master/v3/README.md). Usually, once the cause of hogging is isolated, it is easily fixed.
